app/service/rest_svc.py

```diff
@@ -234,6 +235,23 @@ async def validate(op):
             operation[0].obfuscator = obfuscator
             self.log.debug('Updated operation=%s obfuscator to %s' % (op_id, operation[0].obfuscator))
 
+    async def get_agent_configuration(self, data):
+        plugins = [p for p in await self.get_service('data_svc').locate('plugins', dict(enabled=True)) if p.data_dir]
```

Comment:
We should avoid searching through the file system like this for a couple reasons: 
1) we've already loaded all this data
2) calls to the OS to load files for the most part should live in the data_svc
3) this depends on the file names correctly following the convention of being the ability id (I'm not sure we want to commit to keeping that convention long-term). 

I think you approached it this way because Ability.test kind of annoyingly gives you a base64 encoded test with the contact addresses already baked in -- to make that a little easier to work with, try adding a property like this to the Ability class (return the plaintext command without interpolating in any of the variables):

```python
    ...
    @property
    def raw_command(self):
        return self.decode_bytes(self._test)
```

Then, instead of scanning the file system, you should be able to do something along the lines of: 

```python
        ...
        abilities = await self.get_service('data_svc').locate('abilities', data)
        platforms = {ability.platform: {ability.executor: {'command': ability.raw_command}} for ability in abilities}
        ...
```

Reply:
Above comment resolved in most recent commit where `_test` was being overwritten in `store()`